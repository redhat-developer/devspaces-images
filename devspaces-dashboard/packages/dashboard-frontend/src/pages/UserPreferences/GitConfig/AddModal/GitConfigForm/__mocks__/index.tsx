/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';

import { Props } from '@/pages/UserPreferences/GitConfig/AddModal/GitConfigForm';
import * as GitConfigStore from '@/store/GitConfig';

export class GitConfigForm extends React.PureComponent<Props> {
  render() {
    const { onChange } = this.props;

    return (
      <div>
        <input
          data-testid="submit-invalid-git-config"
          type="button"
          onClick={() =>
            onChange({ user: { name: 'User One' } } as GitConfigStore.GitConfig, false)
          }
        />
        <input
          data-testid="submit-valid-git-config"
          type="button"
          onClick={() =>
            onChange({ user: { email: 'user-1@chetest.com', name: 'User One' } }, true)
          }
        />
      </div>
    );
  }
}
