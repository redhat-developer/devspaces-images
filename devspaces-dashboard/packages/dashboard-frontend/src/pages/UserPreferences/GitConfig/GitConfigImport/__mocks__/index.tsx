/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';

import { Props } from '@/pages/UserPreferences/GitConfig/GitConfigImport';

export class GitConfigImport extends React.PureComponent<Props> {
  render() {
    const { content, onChange } = this.props;

    return (
      <div>
        <input
          data-testid="submit-invalid-git-config"
          type="button"
          value={content}
          onClick={() => onChange('[user]\n\tname = User One\n', false)}
        />
        <input
          data-testid="submit-valid-git-config"
          type="button"
          onClick={() =>
            onChange('[user]\n\tname = User One\n\temail = user-1@chetest.com\n', true)
          }
        />
      </div>
    );
  }
}
