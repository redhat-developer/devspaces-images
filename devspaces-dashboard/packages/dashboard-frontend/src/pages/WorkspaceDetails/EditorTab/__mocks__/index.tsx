/*
 * Copyright (c) 2018-2021 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import common from '@eclipse-che/common';
import React from 'react';
import { Props, State } from '..';
import devfileApi from '../../../../services/devfileApi';
import { Workspace } from '../../../../services/workspace-adapter';

export default class EditorTab extends React.PureComponent<Props, State> {
  cancelChanges: () => void;

  constructor(props: Props) {
    super(props);

    this.state = {
      devfile: {} as devfileApi.Devfile,
      hasChanges: false,
      isDevfileValid: true,
      hasRequestErrors: false,
      currentRequestError: '',
      isExpanded: false,
      showDevfileV2ConfirmationModal: false,
    };

    this.cancelChanges = () => {
      // no-op
    };
  }

  async handleOnSave(workspace: Workspace) {
    try {
      await this.props.onSave(workspace);
    } catch (e) {
      this.setState({
        currentRequestError: common.helpers.errors.getMessage(e),
      });
    }
  }

  render() {
    return (
      <div>
        Fake Editor Tab
        <button onClick={() => this.handleOnSave(this.props.workspace)}>Save</button>
        <button onClick={() => this.props.onDevWorkspaceWarning()}>Warning</button>
        <span data-testid="current-request-error">{this.state.currentRequestError}</span>
      </div>
    );
  }
}
