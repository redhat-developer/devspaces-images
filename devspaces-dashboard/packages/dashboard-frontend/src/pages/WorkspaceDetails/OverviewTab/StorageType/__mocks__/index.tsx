/*
 * Copyright (c) 2018-2021 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';
import { Props, State } from '..';

export default class StorageTypeFormGroup extends React.PureComponent<Props, State> {
  render() {
    return <div>Fake Storage Type Form</div>;
  }
}
