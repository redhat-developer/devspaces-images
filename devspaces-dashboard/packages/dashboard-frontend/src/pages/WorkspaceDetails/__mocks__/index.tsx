/*
 * Copyright (c) 2018-2021 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';
import { Props, State } from '..';

export const mockShowAlert = jest.fn();
export class WorkspaceDetails extends React.PureComponent<Props, State> {
  render() {
    if (!this.props.workspace) {
      return <div>Not found</div>;
    }
    const workspace = this.props.workspace;
    return (
      <div className="WorkspaceDetails">
        Workspace Details Page
        <div data-testid="props-is-loading">{this.props.isLoading}</div>
        <div data-testid="props-old-workspace-path">
          {this.props.oldWorkspaceLocation?.pathname}
        </div>
        <div data-testid="props-show-convert-button">
          {this.props.showConvertButton === true ? 'true' : 'false'}
        </div>
        <div data-testid="props-workspace-id">{this.props.workspace.id}</div>
        <div data-testid="props-workspace-name">{this.props.workspace.name}</div>
        <div data-testid="props-workspaces-link">{this.props.workspacesLink}</div>
        <button onClick={() => this.props.onSave(workspace)}>Save</button>
        <button onClick={() => this.props.onConvert(workspace)}>Convert</button>
      </div>
    );
  }
}
