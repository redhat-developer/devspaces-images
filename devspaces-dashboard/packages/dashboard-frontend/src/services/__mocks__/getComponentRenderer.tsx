/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import { render } from '@testing-library/react';
import React from 'react';
import renderer from 'react-test-renderer';

export * from '@testing-library/react';

export default function <T extends Array<unknown>>(
  getComponent: (..._args: T) => React.ReactElement,
): {
  createSnapshot: (...args: T) => renderer.ReactTestRenderer;
  renderComponent: (...args: T) => {
    reRenderComponent: (..._args: T) => void;
  };
} {
  return {
    createSnapshot: (...args: T) => {
      return renderer.create(getComponent(...args));
    },
    renderComponent: (...args: T) => {
      const res = render(getComponent(...args));
      return {
        reRenderComponent: (..._args: T) => {
          res.rerender(getComponent(..._args));
        },
      };
    },
  };
}
