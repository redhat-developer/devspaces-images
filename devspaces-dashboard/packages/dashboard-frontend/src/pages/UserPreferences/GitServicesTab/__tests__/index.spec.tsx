/*
 * Copyright (c) 2018-2023 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import userEvent from '@testing-library/user-event';
import React from 'react';
import { Provider } from 'react-redux';
import { Store } from 'redux';

import { FakeGitOauthBuilder } from '@/pages/UserPreferences/GitServicesTab/__tests__/__mocks__/gitOauthRowBuilder';
import getComponentRenderer, { screen } from '@/services/__mocks__/getComponentRenderer';
import { FakeStoreBuilder } from '@/store/__mocks__/storeBuilder';
import { actionCreators } from '@/store/GitOauthConfig';
import {
  selectGitOauth,
  selectIsLoading,
  selectProvidersWithToken,
  selectSkipOauthProviders,
} from '@/store/GitOauthConfig/selectors';

import { GitServices } from '..';

// mute the outputs
console.error = jest.fn();

describe('GitServices', () => {
  const mockRevokeOauth = jest.fn();
  const requestGitOauthConfig = jest.fn();
  const requestSkipAuthorizationProviders = jest.fn();
  const deleteSkipOauth = jest.fn();

  const { createSnapshot, renderComponent } = getComponentRenderer(getComponent);

  function getComponent(store: Store): React.ReactElement {
    const state = store.getState();
    const gitOauth = selectGitOauth(state);
    const isLoading = selectIsLoading(state);
    const providersWithToken = selectProvidersWithToken(state);
    const skipOauthProviders = selectSkipOauthProviders(state);
    return (
      <Provider store={store}>
        <GitServices
          gitOauth={gitOauth}
          isLoading={isLoading}
          revokeOauth={mockRevokeOauth}
          deleteSkipOauth={deleteSkipOauth}
          requestGitOauthConfig={requestGitOauthConfig}
          requestSkipAuthorizationProviders={requestSkipAuthorizationProviders}
          providersWithToken={providersWithToken}
          skipOauthProviders={skipOauthProviders}
        />
      </Provider>
    );
  }

  afterEach(() => {
    jest.clearAllMocks();
  });

  describe('without git services', () => {
    let store: Store;

    beforeEach(() => {
      store = new FakeStoreBuilder().build();
    });

    test('snapshot', () => {
      const snapshot = createSnapshot(store);

      expect(snapshot.toJSON()).toMatchSnapshot();
    });

    test('empty state text', () => {
      renderComponent(store);

      const emptyStateText = screen.queryByText('No Git Services');
      expect(emptyStateText).toBeTruthy();
    });
  });

  describe('with 4 git services', () => {
    let store: Store;

    beforeEach(() => {
      store = new FakeStoreBuilder()
        .withGitOauthConfig(
          [
            new FakeGitOauthBuilder()
              .withName('github')
              .withEndpointUrl('https://github.dummy.endpoint.com')
              .build(),
            new FakeGitOauthBuilder()
              .withName('gitlab')
              .withEndpointUrl('https://gitlab.dummy.endpoint.com')
              .build(),
            new FakeGitOauthBuilder()
              .withName('bitbucket')
              .withEndpointUrl('https://bitbucket.dummy.endpoint.org')
              .build(),
            new FakeGitOauthBuilder()
              .withName('azure-devops')
              .withEndpointUrl('https://azure.dummy.endpoint.com/')
              .build(),
          ],
          ['github'],
          [],
        )
        .build();
    });

    test('providers actions depending on authorization state', () => {
      renderComponent(store);

      const emptyStateText = screen.queryByText('No Git Services');
      expect(emptyStateText).not.toBeTruthy();

      const actions = screen.queryAllByRole('button', { name: /actions/i });

      expect(actions.length).toEqual(4);
      expect(actions[0]).not.toBeDisabled();
      expect(actions[1]).toBeDisabled();
      expect(actions[2]).toBeDisabled();
      expect(actions[3]).toBeDisabled();
    });

    test('snapshot', () => {
      const snapshot = createSnapshot(store);

      expect(snapshot.toJSON()).toMatchSnapshot();
    });
  });

  it('should revoke a git service', () => {
    const spyRevokeOauth = jest.spyOn(actionCreators, 'revokeOauth');
    const store = new FakeStoreBuilder()
      .withGitOauthConfig(
        [
          new FakeGitOauthBuilder()
            .withName('github')
            .withEndpointUrl('https://github.com')
            .build(),
        ],
        ['github'],
        [],
      )
      .build();
    renderComponent(store);

    const menuButton = screen.getByLabelText('Actions');
    expect(menuButton).not.toBeDisabled();
    userEvent.click(menuButton);

    const revokeItem = screen.getByRole('menuitem', { name: /Revoke/i });
    userEvent.click(revokeItem);

    const text = screen.findByText("Would you like to revoke git service 'GitHub'?");
    expect(text).toBeTruthy();

    const revokeButton = screen.getByTestId('revoke-button');
    expect(revokeButton).toBeDisabled();

    const checkbox = screen.getByTestId('warning-info-checkbox');
    userEvent.click(checkbox);
    expect(revokeButton).toBeEnabled();

    userEvent.click(revokeButton);
    expect(spyRevokeOauth).toHaveBeenLastCalledWith('github');
  });
});
