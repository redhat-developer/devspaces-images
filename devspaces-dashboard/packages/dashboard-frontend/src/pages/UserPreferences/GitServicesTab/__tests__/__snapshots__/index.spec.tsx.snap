// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GitServices with 4 git services snapshot 1`] = `
[
  <span
    className="progressLine"
  >
    
  </span>,
  <section
    className="pf-c-page__main-section"
  >
    <div
      className="pf-c-toolbar pf-m-page-insets"
      data-ouia-component-id="OUIA-Generated-Toolbar-2"
      data-ouia-component-type="PF4/Toolbar"
      data-ouia-safe={true}
      id="git-services-list-toolbar"
    >
      <div
        className="pf-c-toolbar__content"
      >
        <div
          className="pf-c-toolbar__content-section"
        >
          <div
            className="pf-c-toolbar__item"
          >
            <button
              aria-disabled={true}
              aria-label={null}
              className="pf-c-button pf-m-danger pf-m-disabled"
              data-ouia-component-id="OUIA-Generated-Button-danger-2"
              data-ouia-component-type="PF4/Button"
              data-ouia-safe={true}
              data-testid="bulk-revoke-button"
              disabled={true}
              onClick={[Function]}
              role={null}
              tabIndex={null}
              type="button"
            >
              Revoke
            </button>
          </div>
        </div>
        <div
          className="pf-c-toolbar__expandable-content"
          id="git-services-list-toolbar-expandable-content-1"
        >
          <div
            className="pf-c-toolbar__group"
          />
        </div>
      </div>
      <div
        className="pf-c-toolbar__content pf-m-hidden"
        hidden={true}
      >
        <div
          className="pf-c-toolbar__group"
        />
      </div>
    </div>
    <table
      aria-label="Git services"
      className="pf-c-table pf-m-grid-md pf-m-compact"
      data-ouia-component-id="OUIA-Generated-Table-4"
      data-ouia-component-type="PF4/Table"
      data-ouia-safe={true}
      role="grid"
    >
      <thead
        className=""
      >
        <tr
          className=""
          data-ouia-component-id="OUIA-Generated-TableRow-6"
          data-ouia-component-type="PF4/TableRow"
          data-ouia-safe={true}
          hidden={false}
        >
          <th
            className=""
            data-key={0}
            data-label=""
            onMouseEnter={[Function]}
            scope={null}
          >
            
          </th>
          <th
            className=""
            data-key={1}
            data-label="Name"
            onMouseEnter={[Function]}
            scope="col"
          >
            Name
          </th>
          <th
            className=""
            data-key={2}
            data-label="Server"
            onMouseEnter={[Function]}
            scope="col"
          >
            Server
          </th>
          <th
            className=""
            data-key={3}
            data-label="Authorization"
            onMouseEnter={[Function]}
            scope="col"
          >
            Authorization
          </th>
          <td
            className=""
            data-key={4}
            data-label=""
            onMouseEnter={[Function]}
            scope={null}
          >
            
          </td>
        </tr>
      </thead>
      <tbody
        className=""
        role="rowgroup"
      >
        <tr
          className=""
          data-ouia-component-id="OUIA-Generated-TableRow-7"
          data-ouia-component-type="PF4/TableRow"
          data-ouia-safe={true}
          hidden={false}
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          <td
            className="pf-c-table__check"
            data-key={0}
            data-label={null}
            onMouseEnter={[Function]}
          >
            <label>
              <input
                aria-label="Select row 0"
                checked={false}
                name="checkrow0"
                onChange={[Function]}
                type="checkbox"
              />
            </label>
          </td>
          <td
            className=""
            data-key={1}
            data-label="Name"
            onMouseEnter={[Function]}
          >
            GitHub
          </td>
          <td
            className=""
            data-key={2}
            data-label="Server"
            onMouseEnter={[Function]}
          >
            <a
              href="https://github.dummy.endpoint.com"
              rel="noreferrer"
              target="_blank"
            >
              https://github.dummy.endpoint.com
            </a>
          </td>
          <td
            className=""
            data-key={3}
            data-label="Authorization"
            onMouseEnter={[Function]}
          >
            <div>
              <svg
                aria-hidden={true}
                aria-labelledby={null}
                fill="var(--pf-global--success-color--100)"
                height="1em"
                role="img"
                style={
                  {
                    "verticalAlign": "-0.125em",
                  }
                }
                viewBox="0 0 512 512"
                width="1em"
              >
                <path
                  d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
                />
              </svg>
              <span>
                User has been authorized successfully.
              </span>
            </div>
          </td>
          <td
            className="pf-c-table__action"
            data-key={4}
            data-label={null}
            onMouseEnter={[Function]}
            style={
              {
                "paddingRight": 0,
              }
            }
          >
            <div
              className="pf-c-dropdown pf-m-align-right"
              data-ouia-component-id="OUIA-Generated-Dropdown-5"
              data-ouia-component-type="PF4/Dropdown"
              data-ouia-safe={true}
            >
              <button
                aria-expanded={false}
                aria-haspopup={true}
                aria-label="Actions"
                className="pf-c-dropdown__toggle pf-m-plain"
                disabled={false}
                id="pf-dropdown-toggle-id-4"
                onClick={[Function]}
                onKeyDown={[Function]}
                type="button"
              >
                <svg
                  aria-hidden={true}
                  aria-labelledby={null}
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style={
                    {
                      "verticalAlign": "-0.125em",
                    }
                  }
                  viewBox="0 0 192 512"
                  width="1em"
                >
                  <path
                    d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
                  />
                </svg>
              </button>
            </div>
          </td>
        </tr>
        <tr
          className=""
          data-ouia-component-id="OUIA-Generated-TableRow-8"
          data-ouia-component-type="PF4/TableRow"
          data-ouia-safe={true}
          hidden={false}
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          <td
            className="pf-c-table__check"
            data-key={0}
            data-label={null}
            onMouseEnter={[Function]}
          >
            <label>
              <input
                aria-label="Select row 1"
                checked={false}
                disabled={true}
                name="checkrow1"
                onChange={[Function]}
                type="checkbox"
              />
            </label>
          </td>
          <td
            className=""
            data-key={1}
            data-label="Name"
            onMouseEnter={[Function]}
          >
            GitLab
          </td>
          <td
            className=""
            data-key={2}
            data-label="Server"
            onMouseEnter={[Function]}
          >
            <a
              href="https://gitlab.dummy.endpoint.com"
              rel="noreferrer"
              target="_blank"
            >
              https://gitlab.dummy.endpoint.com
            </a>
          </td>
          <td
            className=""
            data-key={3}
            data-label="Authorization"
            onMouseEnter={[Function]}
          >
            <div>
              <svg
                aria-hidden={true}
                aria-labelledby={null}
                fill="var(--pf-global--disabled-color--100)"
                height="1em"
                role="img"
                style={
                  {
                    "verticalAlign": "-0.125em",
                  }
                }
                viewBox="0 0 1024 1024"
                width="1em"
              >
                <path
                  d="M512,896 C300.2,896 128,723.9 128,512 C128,300.3 300.2,128 512,128 C723.7,128 896,300.2 896,512 C896,723.8 723.7,896 512,896 L512,896 Z M512.1,0 C229.7,0 0,229.8 0,512 C0,794.3 229.8,1024 512.1,1024 C794.4,1024 1024,794.3 1024,512 C1024,229.7 794.4,0 512.1,0 L512.1,0 Z"
                />
              </svg>
              <span>
                User has not been authorized yet.
              </span>
            </div>
          </td>
          <td
            className="pf-c-table__action"
            data-key={4}
            data-label={null}
            onMouseEnter={[Function]}
            style={
              {
                "paddingRight": 0,
              }
            }
          >
            <div
              className="pf-c-dropdown pf-m-align-right"
              data-ouia-component-id="OUIA-Generated-Dropdown-6"
              data-ouia-component-type="PF4/Dropdown"
              data-ouia-safe={true}
            >
              <button
                aria-expanded={false}
                aria-haspopup={true}
                aria-label="Actions"
                className="pf-c-dropdown__toggle pf-m-plain"
                disabled={true}
                id="pf-dropdown-toggle-id-5"
                onClick={[Function]}
                onKeyDown={[Function]}
                type="button"
              >
                <svg
                  aria-hidden={true}
                  aria-labelledby={null}
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style={
                    {
                      "verticalAlign": "-0.125em",
                    }
                  }
                  viewBox="0 0 192 512"
                  width="1em"
                >
                  <path
                    d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
                  />
                </svg>
              </button>
            </div>
          </td>
        </tr>
        <tr
          className=""
          data-ouia-component-id="OUIA-Generated-TableRow-9"
          data-ouia-component-type="PF4/TableRow"
          data-ouia-safe={true}
          hidden={false}
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          <td
            className="pf-c-table__check"
            data-key={0}
            data-label={null}
            onMouseEnter={[Function]}
          >
            <label>
              <input
                aria-label="Select row 2"
                checked={false}
                disabled={true}
                name="checkrow2"
                onChange={[Function]}
                type="checkbox"
              />
            </label>
          </td>
          <td
            className=""
            data-key={1}
            data-label="Name"
            onMouseEnter={[Function]}
          >
            Bitbucket Server (OAuth 1.0)
          </td>
          <td
            className=""
            data-key={2}
            data-label="Server"
            onMouseEnter={[Function]}
          >
            <a
              href="https://bitbucket.dummy.endpoint.org"
              rel="noreferrer"
              target="_blank"
            >
              https://bitbucket.dummy.endpoint.org
            </a>
          </td>
          <td
            className=""
            data-key={3}
            data-label="Authorization"
            onMouseEnter={[Function]}
          >
            <div>
              <svg
                aria-hidden={true}
                aria-labelledby={null}
                fill="var(--pf-global--disabled-color--100)"
                height="1em"
                role="img"
                style={
                  {
                    "verticalAlign": "-0.125em",
                  }
                }
                viewBox="0 0 1024 1024"
                width="1em"
              >
                <path
                  d="M512,896 C300.2,896 128,723.9 128,512 C128,300.3 300.2,128 512,128 C723.7,128 896,300.2 896,512 C896,723.8 723.7,896 512,896 L512,896 Z M512.1,0 C229.7,0 0,229.8 0,512 C0,794.3 229.8,1024 512.1,1024 C794.4,1024 1024,794.3 1024,512 C1024,229.7 794.4,0 512.1,0 L512.1,0 Z"
                />
              </svg>
              <span>
                User has not been authorized yet.
              </span>
            </div>
          </td>
          <td
            className="pf-c-table__action"
            data-key={4}
            data-label={null}
            onMouseEnter={[Function]}
            style={
              {
                "paddingRight": 0,
              }
            }
          >
            <div
              className="pf-c-dropdown pf-m-align-right"
              data-ouia-component-id="OUIA-Generated-Dropdown-7"
              data-ouia-component-type="PF4/Dropdown"
              data-ouia-safe={true}
            >
              <button
                aria-expanded={false}
                aria-haspopup={true}
                aria-label="Actions"
                className="pf-c-dropdown__toggle pf-m-plain"
                disabled={true}
                id="pf-dropdown-toggle-id-6"
                onClick={[Function]}
                onKeyDown={[Function]}
                type="button"
              >
                <svg
                  aria-hidden={true}
                  aria-labelledby={null}
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style={
                    {
                      "verticalAlign": "-0.125em",
                    }
                  }
                  viewBox="0 0 192 512"
                  width="1em"
                >
                  <path
                    d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
                  />
                </svg>
              </button>
            </div>
          </td>
        </tr>
        <tr
          className=""
          data-ouia-component-id="OUIA-Generated-TableRow-10"
          data-ouia-component-type="PF4/TableRow"
          data-ouia-safe={true}
          hidden={false}
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          <td
            className="pf-c-table__check"
            data-key={0}
            data-label={null}
            onMouseEnter={[Function]}
          >
            <label>
              <input
                aria-label="Select row 3"
                checked={false}
                disabled={true}
                name="checkrow3"
                onChange={[Function]}
                type="checkbox"
              />
            </label>
          </td>
          <td
            className=""
            data-key={1}
            data-label="Name"
            onMouseEnter={[Function]}
          >
            Microsoft Azure DevOps
          </td>
          <td
            className=""
            data-key={2}
            data-label="Server"
            onMouseEnter={[Function]}
          >
            <a
              href="https://azure.dummy.endpoint.com/"
              rel="noreferrer"
              target="_blank"
            >
              https://azure.dummy.endpoint.com/
            </a>
          </td>
          <td
            className=""
            data-key={3}
            data-label="Authorization"
            onMouseEnter={[Function]}
          >
            <div>
              <svg
                aria-hidden={true}
                aria-labelledby={null}
                fill="var(--pf-global--disabled-color--100)"
                height="1em"
                role="img"
                style={
                  {
                    "verticalAlign": "-0.125em",
                  }
                }
                viewBox="0 0 1024 1024"
                width="1em"
              >
                <path
                  d="M512,896 C300.2,896 128,723.9 128,512 C128,300.3 300.2,128 512,128 C723.7,128 896,300.2 896,512 C896,723.8 723.7,896 512,896 L512,896 Z M512.1,0 C229.7,0 0,229.8 0,512 C0,794.3 229.8,1024 512.1,1024 C794.4,1024 1024,794.3 1024,512 C1024,229.7 794.4,0 512.1,0 L512.1,0 Z"
                />
              </svg>
              <span>
                User has not been authorized yet.
              </span>
            </div>
          </td>
          <td
            className="pf-c-table__action"
            data-key={4}
            data-label={null}
            onMouseEnter={[Function]}
            style={
              {
                "paddingRight": 0,
              }
            }
          >
            <div
              className="pf-c-dropdown pf-m-align-right"
              data-ouia-component-id="OUIA-Generated-Dropdown-8"
              data-ouia-component-type="PF4/Dropdown"
              data-ouia-safe={true}
            >
              <button
                aria-expanded={false}
                aria-haspopup={true}
                aria-label="Actions"
                className="pf-c-dropdown__toggle pf-m-plain"
                disabled={true}
                id="pf-dropdown-toggle-id-7"
                onClick={[Function]}
                onKeyDown={[Function]}
                type="button"
              >
                <svg
                  aria-hidden={true}
                  aria-labelledby={null}
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style={
                    {
                      "verticalAlign": "-0.125em",
                    }
                  }
                  viewBox="0 0 192 512"
                  width="1em"
                >
                  <path
                    d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
                  />
                </svg>
              </button>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </section>,
]
`;

exports[`GitServices without git services snapshot 1`] = `
[
  <span
    className="progressLine"
  >
    
  </span>,
  <section
    className="pf-c-page__main-section"
  >
    <div
      className="pf-c-empty-state pf-m-sm pf-m-full-height"
    >
      <div
        className="pf-c-empty-state__content"
      >
        <svg
          aria-hidden="true"
          aria-labelledby={null}
          className="pf-c-empty-state__icon"
          fill="currentColor"
          height="1em"
          role="img"
          style={
            {
              "verticalAlign": "-0.125em",
            }
          }
          viewBox="0 0 1152 1024"
          width="1em"
        >
          <path
            d="M504.601506,583.6 C499.901506,576.8 493.701506,572 485.901506,569.1 L270.001506,450.5 C265.501506,448.8 261.001506,448 256.401506,448 L255.701506,448 L255.001506,448 C250.601506,448 246.001506,448.8 241.601506,450.5 L25.7015061,569.2 C18.0015061,572.1 11.7015061,576.9 7.00150606,583.7 C2.30150606,590.4 0.00074731781,598 0.00074731781,606.2 L0.00074731781,862.6 C-0.098493943,879.8 9.70150606,886 20.4015061,892 L236.401506,1019.1 C242.101506,1022.4 248.401506,1024 255.101506,1024 L255.801506,1024 L256.501506,1024 C263.201506,1024 269.501506,1022.4 275.201506,1019.1 L491.201506,892 C501.401506,886.5 511.701506,879.8 511.701506,865.4 L511.701506,606.1 C511.701506,597.9 509.301506,590.4 504.601506,583.6 Z M474.801506,832 L255.801506,960 L35.6015061,832 L35.6015061,768 L255.701506,896 L474.801506,768 L474.801506,832 Z M474.801506,704 L255.801506,832 L35.6015061,704 L35.6015061,640 L255.701506,768 L474.801506,640 L474.801506,704 Z M1144.60151,583.6 C1139.90151,576.8 1133.70151,572 1125.90151,569.1 L910.001506,450.5 C905.501506,448.8 901.001506,448 896.401506,448 L895.701506,448 L895.001506,448 C890.601506,448 886.001506,448.8 881.601506,450.5 L665.701506,569.2 C658.001506,572.1 651.701506,576.9 647.001506,583.7 C642.301506,590.4 640.001506,598 640.001506,606.2 L640.001506,862.6 C640.001506,879.8 649.801506,886 660.501506,892 L876.501506,1019.1 C882.201506,1022.4 888.501506,1024 895.201506,1024 L895.901506,1024 L896.601506,1024 C903.301506,1024 909.601506,1022.4 915.301506,1019.1 L1131.30151,892 C1141.50151,886.5 1151.80151,879.8 1151.80151,865.4 L1151.80151,606.1 C1151.70151,597.9 1149.30151,590.4 1144.60151,583.6 Z M1114.80151,832 L895.701506,960 L675.601506,832 L675.601506,768 L895.701506,896 L1114.80151,768 L1114.80151,832 Z M1114.80151,704 L895.701506,832 L675.601506,704 L675.601506,640 L895.701506,768 L1114.80151,640 L1114.80151,704 Z M824.601506,95.6 C819.901506,88.8 813.701506,84 805.901506,81.1 L590.001506,2.5 C585.501506,0.8 581.001506,0 576.401506,0 L575.701506,0 L575.001506,0 C570.601506,0 566.001506,0.8 561.601506,2.5 L345.701506,81.2 C338.001506,84.1 331.701506,88.9 327.001506,95.7 C322.401506,102.4 320.001506,110 320.001506,118.2 L320.001506,359.6 C320.001506,376.8 329.801506,383 340.501506,389 L556.501506,507.1 C562.201506,510.4 568.501506,512 575.201506,512 L575.901506,512 L576.601506,512 C583.301506,512 589.601506,510.4 595.301506,507.1 L811.301506,389 C821.501506,383.5 831.801506,376.8 831.801506,362.4 L831.801506,118.1 C831.701506,109.9 829.301506,102.4 824.601506,95.6 Z M794.801506,332.3 L575.801506,448 L355.601506,332.3 L355.601506,272 L575.701506,384 L794.801506,272 L794.801506,332.3 Z M794.801506,208 L575.801506,320 L355.601506,208 L355.601506,144 L575.701506,256 L794.801506,144 L794.801506,208 Z"
          />
        </svg>
        <h4
          className="pf-c-title pf-m-lg"
          data-ouia-component-id="OUIA-Generated-Title-1"
          data-ouia-component-type="PF4/Title"
          data-ouia-safe={true}
        >
          No Git Services
        </h4>
      </div>
    </div>
  </section>,
]
`;
